pilot/pkg/networking/core/v1alpha3/listener_test.go

```diff
@@ -164,6 +164,10 @@ func TestOutboundListenerConflict_TCPWithCurrentTCP(t *testing.T) {
 }
 
 func TestOutboundListenerTCPWithVS(t *testing.T) {
+	_ = os.Setenv("PILOT_ENABLE_FALLTHROUGH_ROUTE", "false")
```

Comment:
why did this need to change?

Reply:
The test will fail without this env var. Someone added `appendListenerFallthroughRoute` which will increase the number of filter chain number by one but didn't fix the test...

Reply:
this test passed before this PR, so that seems to indicate your PR changed behavior and is not just a refactoring?

Reply:
No. I pulled the latest master branch and the test failed before I starting working on refactoring.

Reply:
we probably set this env variable in a different test and don't unset it. So we should fix the other test as well.

Reply:
Yes. It should be fixed by another pr.